src/components/Popups/PopupItem.tsx

```diff
@@ -45,32 +48,32 @@ export default function PopupItem({
   popKey: string
 }) {
   const removePopup = useRemovePopup()
-  const removeThisPopup = useCallback(() => removePopup(popKey), [popKey, removePopup])
+  const theme = useTheme()
+
   useEffect(() => {
     if (removeAfterMs === null) return undefined
 
     const timeout = setTimeout(() => {
-      removeThisPopup()
+      removePopup(popKey)
     }, removeAfterMs)
 
     return () => {
       clearTimeout(timeout)
     }
-  }, [removeAfterMs, removeThisPopup])
-
-  const theme = useTheme()
+  }, [popKey, removeAfterMs, removePopup])
+  const isTxnPopup = 'txn' in content
 
   let popupContent
-  if ('txn' in content) {
+  if (isTxnPopup) {
     popupContent = <TransactionPopup hash={content.txn.hash} />
   } else if ('failedSwitchNetwork' in content) {
     popupContent = <FailedNetworkSwitchPopup chainId={content.failedSwitchNetwork} />
   }
 
-  return popupContent ? (
-    <Popup>
-      <StyledClose color={theme.textSecondary} onClick={removeThisPopup} />
+  return (
+    <Popup isTransactionPopup={isTxnPopup} show={!!popupContent}>
+      <StyledClose padding={isTxnPopup ? 16 : 20} color={theme.textSecondary} onClick={() => removePopup(popKey)} />
```

Comment:
Why not pass this as a Boolean too? It's inconsistent with its wrapper.

Reply:
done